app/src/main/java/org/schabi/newpipe/NewVersionWorker.kt

```diff
@@ -0,0 +1,122 @@
+package org.schabi.newpipe
+
+import android.content.Context
+import android.util.Log
+import androidx.core.content.edit
+import androidx.preference.PreferenceManager
+import androidx.work.OneTimeWorkRequest
+import androidx.work.WorkManager
+import androidx.work.WorkRequest
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.grack.nanojson.JsonParser
+import com.grack.nanojson.JsonParserException
+import org.schabi.newpipe.extractor.downloader.Response
+import org.schabi.newpipe.extractor.exceptions.ReCaptchaException
+import org.schabi.newpipe.util.ReleaseUtil.coerceExpiry
+import org.schabi.newpipe.util.ReleaseUtil.compareAppVersionAndShowNotification
+import org.schabi.newpipe.util.ReleaseUtil.isExpired
+import org.schabi.newpipe.util.ReleaseUtil.isReleaseApk
+import java.io.IOException
+
+class NewVersionWorker(
+    context: Context,
+    workerParams: WorkerParameters
+) : Worker(context, workerParams) {
+
+    @Throws(IOException::class, ReCaptchaException::class)
+    private fun checkNewVersion() {
+        // Check if the current apk is a github one or not.
+        if (!isReleaseApk()) {
+            return
+        }
+
+        val prefs = PreferenceManager.getDefaultSharedPreferences(applicationContext)
+        // Check if the last request has happened a certain time ago
+        // to reduce the number of API requests.
+        val expiry = prefs.getLong(applicationContext.getString(R.string.update_expiry_key), 0)
+        if (!isExpired(expiry)) {
+            return
+        }
+
+        // Make a network request to get latest NewPipe data.
+        val response = DownloaderImpl.getInstance()[NEWPIPE_API_URL]
```

Comment:
[I understand](https://kotlinlang.org/docs/operator-overloading.html#indexed-access-operator) that Kotlin allows writing `[]` instead of `.get()`, but I'd use `.get()` here since it's not accessing a list ;-)
```suggestion
        val response = DownloaderImpl.getInstance().get(NEWPIPE_API_URL)
```

Reply:
Sure 👍